### PR TITLE
Fixed loading translation values for User Profile attributes

### LINKED ISSUE #27837: Translation values not loaded for User Profile attributes
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

The dialog for user profile attribute translations does not load the current values of the translations.

There are two ways to see this bug as described below.

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

Translations will be loaded when the dialog is shown.

### Actual behavior

Translations are always blank when you first open the dialog.

### How to Reproduce?

**Scenario 1**
1. Create UP attribute
2. Open dialog to enter translations for the attribute description
3. Save (translations are properly saved to realm overrides)
4. Edit attribute again.
5. Open dialog to edit translations.
6. See that translations are blank.

**Scenario 2**
1. Create UP attribute
2. Open dialog to enter translations for the attribute description
3. Don't save
4. Open dialog again.
5. See that translations are blank.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by agagancarczyk**
Closes https://github.com/keycloak/keycloak/issues/27837

### CODE REVIEW

**Review by jonkoops — DISMISSED** [review]

File: js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx [hunk 1]
```diff
@@ -105,15 +108,60 @@ export const AddTranslationsDialog = ({
     );
   }, [combinedLocales, filter, whoAmI]);
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (agagancarczyk) [reply js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx]: sounds good @jonkoops

**Review by ssilvert — CHANGES_REQUESTED** [review]
I tried the code out and it's not working.  When I click on the button to bring up the translations dialog I get a NullPointerException on the server.  I am using the latest nightly build of the server.
`Caused by: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.UserSessionModel.getUser()" because "userSession" is null`

The request that is failing is `http://localhost:8180/admin/realms/test/localization/en`.
500 Internal Server Error

@agagancarczyk Maybe try this with the latest nightly and see if you see this too?

**Comment by jonkoops** [comment]
@ssilvert I've also been seeing that error locally in other places, it is related how things are proxied during development. It should work if you do a full production build and test against that. Unfortunately, this also makes it impossible to run the Cypress tests on some suites against the development server.

I think we should unify the dev server and the Keycloak server, so that the Keycloak server always is responsible for serving the front-end, even during development. I've started #27311 as a thought experiment on how to accomplish this.

**Comment by ssilvert** [comment]
@agagancarczyk @jonkoops OK, I was able to run the code by doing a full build and scenario 1 is fixed.

I am still seeing scenario 2 from the issue.  Agi, note that I updated the steps for scenario 2 as I was unclear in my original post.

I also added scenario 3, which is another related problem.  See the issue.

**Comment by ssilvert** [comment]
> I think we should unify the dev server and the Keycloak server, so that the Keycloak server always is responsible for serving the front-end, even during development. I've started #27311 as a thought experiment on how to accomplish this.

@jonkoops That would be amazing if you can pull it off.  It would also solve some other problems I've seen such as changes to message bundles that are not being reflected in dev mode.

**Comment by jonkoops** [comment]
I'll try to put some time into it when we're done with the PatternFly migration :+1:

**Comment by agagancarczyk** [comment]
> @agagancarczyk @jonkoops OK, I was able to run the code by doing a full build and scenario 1 is fixed.
> 
> I am still seeing scenario 2 from the issue. Agi, note that I updated the steps for scenario 2 as I was unclear in my original post.
> 
> I also added scenario 3, which is another related problem. See the issue.

@ssilvert scenario 2 has also been fixed. Can you try again? I pushed a fix yesterday. Maybe you are expecting something else? I'll take a look at scenario 3.

**Comment by agagancarczyk** [comment]
Closing this PR as it's old in favour of https://github.com/keycloak/keycloak/pull/28899

**Pull request closed without merging**